### PR TITLE
feat: add Tilt support for local development

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
This will allow for more rapid development as tilt supports
hot reload on code changes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
No open issues currently

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -22,8 +22,6 @@ spec:
       labels:
         control-plane: controller-manager
     spec:
-      securityContext:
-        runAsNonRoot: true
```
Comment (joekr) [inline config/manager/manager.yaml:26]: This was causing errors in the development environment and I noticed no other provider was setting this for their manager.yaml.

If we don't want to remove this no worries, I'll try to find a way with it `true`

**Comment by joekr** [comment]
# unit tests

```
?       github.com/oracle/cluster-api-provider-oci      [no test files]
ok      github.com/oracle/cluster-api-provider-oci/api/v1beta1  0.251s  coverage: 18.7% of statements
?       github.com/oracle/cluster-api-provider-oci/cloud/config [no test files]
ok      github.com/oracle/cluster-api-provider-oci/cloud/ociutil        0.208s  coverage: 25.0% of statements
ok      github.com/oracle/cluster-api-provider-oci/cloud/scope  176.020s        coverage: 73.4% of statements
?       github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks    [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/compute       [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute  [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement     [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/identity      [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity        [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer   [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb  [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/vcn   [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn  [no test files]
ok      github.com/oracle/cluster-api-provider-oci/controllers  26.255s coverage: 71.0% of statements
?       github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1      [no test files]
?       github.com/oracle/cluster-api-provider-oci/exp/controllers      [no test files]
?       github.com/oracle/cluster-api-provider-oci/feature      [no test files]
```

# e2e tests

```
```

**Review by arindam-bandyopadhyay — APPROVED** [review]
LGTM

File: local-dev.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+```
+cat <<EOF > tilt-settings.json
+{
+  "kustomize_substitutions": {
+      "OCI_TENANCY_ID_B64": "$(echo "${OCI_TENANCY_ID}" | tr -d '\n' | base64 | tr -d '\n')",
+      "OCI_CREDENTIALS_FINGERPRINT_B64": "$(echo "${OCI_CREDENTIALS_FINGERPRINT}" | tr -d '\n' | base64 | tr -d '\n')",
+      "OCI_USER_ID_B64": "$(echo "${OCI_USER_ID}" | tr -d '\n' | base64 | tr -d '\n')",
+      "OCI_REGION_B64": "$(echo "${OCI_REGION}" | tr -d '\n' | base64 | tr -d '\n')",
+      "OCI_CREDENTIALS_KEY_B64": "$(echo "${OCI_CREDENTIALS_KEY_PATH}" | tr -d '\n' | base64 | tr -d '\n')"
```
Comment (arindam-bandyopadhyay) [inline local-dev.md:37]: We don't have `OCI_CREDENTIALS_KEY_PATH` in https://oracle.github.io/cluster-api-provider-oci/gs/install-cluster-api.html#configure-authentication

**Review by arindam-bandyopadhyay — APPROVED** [review]

**Pull request merged**